impl/src/main/java/com/sun/faces/el/EmptyStringToNullELResolver.java

```diff
@@ -32,7 +34,7 @@ public Class<?> getCommonPropertyType(ELContext context, Object base) {
     @Override
     @SuppressWarnings("unchecked")
     public <T> T convertToType(ELContext context, Object value, Class<T> targetType) {
-        if (value == null && targetType == String.class) {
+        if ( value == null && targetType == String.class && EVALUATION_CONTEXT_CLASS_SIMPLE_NAME.equals(context.getClass().getSimpleName()) ) {
```

Comment:
This is a hack not a fix. This is prone to fail when a different EL implementation is used which doesn't use `EvaluationContext` as class name in this specific situation.

The correct fix is most probably to just return an empty string.

Reply:
Yes I know :smiley:  :smiley: 
but at the moment I had to fix my fork, and it works at least on Tomcat  ...
I'm not able to fix it on a long term perspective :smiling_face_with_tear:

Reply:
This hack goes wrong on `#{bean.setValue(null)}`. It will unexpectedly set empty string there.